chain/client/src/chunk_inclusion_tracker.rs

```diff
@@ -2,25 +2,36 @@ use itertools::Itertools;
 use lru::LruCache;
 use near_async::time::Utc;
 use near_chain_primitives::Error;
+use near_o11y::log_assert_fail;
 use near_primitives::block_body::ChunkEndorsementSignatures;
 use near_primitives::hash::CryptoHash;
 use near_primitives::sharding::{ChunkHash, ShardChunkHeader};
 use near_primitives::types::{AccountId, EpochId, ShardId};
 use std::collections::HashMap;
 
 use crate::metrics;
-use crate::stateless_validation::chunk_endorsement_tracker::ChunkEndorsementTracker;
+use crate::stateless_validation::chunk_endorsement_tracker::{
+    ChunkEndorsementTracker, ChunkEndorsementsState,
+};
 
 const CHUNK_HEADERS_FOR_INCLUSION_CACHE_SIZE: usize = 2048;
 const NUM_EPOCH_CHUNK_PRODUCERS_TO_KEEP_IN_BLOCKLIST: usize = 1000;
 
 // chunk_header, received_time and chunk_producer are populated when we call mark_chunk_header_ready_for_inclusion
-// signatures is populated later during call to prepare_chunk_headers_ready_for_inclusion
+// endorsements is populated later during call to prepare_chunk_headers_ready_for_inclusion
 struct ChunkInfo {
     pub chunk_header: ShardChunkHeader,
     pub received_time: Utc,
     pub chunk_producer: AccountId,
-    pub signatures: Option<ChunkEndorsementSignatures>,
+    pub endorsements: Option<ChunkEndorsementsState>,
+}
+
+impl ChunkInfo {
+    fn has_chunk_endorsements(&self) -> bool {
```

Comment:
"is_endorsed" because Endorsed state indicates not just that endorsements are present, but also that chunk has enough stake?
I'm also not fully sure that Option type is needed for `endorsements`, because None looks the same as NotEnoughStake. But I may miss something.